dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -231,7 +164,138 @@ public void close() throws IOException {
 
         @Override
         public String toString() {
-            return "CloseableClient{client=" + client + '}';
+            return "CloseableClient{client="
+                    + client + ", leakDetector="
+                    + leakDetector + ", executor="
+                    + executor + '}';
+        }
+    }
+
+    public static ClientBuilder clientBuilder() {
+        return new ClientBuilder();
+    }
+
+    public static final class ClientBuilder {
+
+        @Nullable
+        private ClientConfiguration clientConfiguration;
+
+        @Nullable
+        private String clientName;
+
+        @Nullable
+        private ExecutorService executor;
+
+        private ClientBuilder() {}
+
+        public ClientBuilder clientConfiguration(ClientConfiguration value) {
+            this.clientConfiguration = Preconditions.checkNotNull(value, "ClientConfiguration is required");
+            return this;
+        }
+
+        public ClientBuilder clientName(String value) {
```

Comment:
Also I remember @mattsills was asking about the significance of this value. Possibly worth some javadoc explaining this is for metrics and doesn't affect requests?

Also needs to be safe loggable!